pkg/cluster/deploybaseresources_additional.go

```diff
@@ -92,6 +93,81 @@ func (m *manager) clusterServicePrincipalRBAC() *arm.Resource {
 	)
 }
 
+func (m *manager) ensureWorkloadIdentityRBAC() ([]*arm.Resource, error) {
+	if !m.doc.OpenShiftCluster.UsesWorkloadIdentity() {
+		return nil, nil
+	}
+
+	clusterMSI, err := m.doc.OpenShiftCluster.ClusterMsiUserAssignedIdentity()
+	if err != nil {
+		return nil, err
+	}
+
+	subnetResource, err := azure.ParseResourceID(m.doc.OpenShiftCluster.Properties.MasterProfile.SubnetID)
+	if err != nil {
+		return nil, err
+	}
+
+	resources := []*arm.Resource{}
+	vnetRG := subnetResource.ResourceGroup
+	managedRG := stringutils.LastTokenByte(m.doc.OpenShiftCluster.Properties.ClusterProfile.ResourceGroupID, '/')
+
+	resources = append(resources, m.workloadIdentityResourceGroupRBAC(rbac.RoleAzureRedHatOpenShiftFederatedCredentialRole, clusterMSI.PrincipalID, managedRG))
+	resources = append(resources, m.workloadIdentityResourceGroupRBAC(rbac.RoleAzureRedHatOpenShiftFederatedCredentialRole, clusterMSI.PrincipalID, vnetRG))
+
+	r, err := m.platoformWorkloadIdentityRBAC(vnetRG, managedRG)
+	if err != nil {
+		return nil, err
+	}
+
+	resources = append(resources, r...)
+	return resources, nil
+}
+
+func (m *manager) platoformWorkloadIdentityRBAC(vnetRG, managedRG string) ([]*arm.Resource, error) {
+	if !m.doc.OpenShiftCluster.UsesWorkloadIdentity() {
+		return nil, nil
+	}
+
+	resources := []*arm.Resource{}
+	platformWIRolesByRoleName := m.platformWorkloadIdentityRolesByVersion.GetPlatformWorkloadIdentityRolesByRoleName()
+	platformWorkloadIdentities := m.doc.OpenShiftCluster.Properties.PlatformWorkloadIdentityProfile.PlatformWorkloadIdentities
+
+	for _, identity := range platformWorkloadIdentities {
+		role, exists := platformWIRolesByRoleName[identity.OperatorName]
+		if !exists {
+			return nil, &api.CloudError{
+				StatusCode: http.StatusBadRequest,
+				CloudErrorBody: &api.CloudErrorBody{
+					Code:    api.CloudErrorCodePlatformWorkloadIdentityMismatch,
+					Message: "OperatorName " + identity.OperatorName + " not found in platform workload identity roles",
+				},
+			}
+		}
+
+		roleID := stringutils.LastTokenByte(role.RoleDefinitionID, '/')
+		resources = append(resources, m.workloadIdentityResourceGroupRBAC(roleID, identity.ObjectID, managedRG))
+
+		if role.OperatorName != "StorageOperator" {
+			resources = append(resources, m.workloadIdentityResourceGroupRBAC(roleID, identity.ObjectID, vnetRG))
```

Comment:
Why are we handling the RBAC for vnetRG? The RBAC is only needed for the managedRG, FPSP won't have permissions for role assignment on vnetRG. And for managedRG, all the identities existing in `platformWIRolesByRoleName` will have a role assignment performed for their corresponding roles.

Reply:
Yes, the VNET is to be handled by customer, and that check will already done till this point as part of dynamic validation.